archiver/service/archiver.go

```diff
@@ -63,6 +66,8 @@ func (a *Archiver) Start(ctx context.Context) error {
 		return err
 	}
 
+	a.waitObtainStorageLock(ctx)
```

Comment:
If we want to add a feature flag to have this functionality turned off by default, we would only need to guard this one line of code with the flag

Reply:
I'm open to adding a feature flag, I don't think it's required though as it shouldn't cause issues to do this. If you want to flag it though -- no objections from me.

Reply:
I'll leave it without a flag as I think the feature now has sufficient test coverage to give confidence that it will not cause problems